### PR TITLE
Add missing Buffer APIs; update CI: node and actions versions

### PR DESCRIPTION
**Pull request by JordanMartinez**
**Description of the change**

Adds more APIs. This may not cover all the APIs but it does add a few more.

---

**Checklist:**

- [ ] Added the change to the changelog's "Unreleased" section with a reference to this PR (e.g. "- Made a change (#0000)")
- [ ] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation
- [ ] Added a test for the contribution (if applicable)

### CODE REVIEW

**Comment by JordanMartinez** [comment]
@thomashoneyman :ping_pong:

File: src/Node/Buffer.purs [hunk 1]
```diff
@@ -81,8 +92,27 @@ usingFromImmutable f buf = f <$> unsafeFreeze buf
...
 create :: Int -> Effect Buffer
-create = usingToImmutable Immutable.create
+create = alloc
+
+-- | Creates a new buffer of the specified size.
+alloc :: Int -> Effect Buffer
+alloc = usingToImmutable Immutable.alloc
+
+-- | Creates a new buffer of the specified size. Unsafe because it reuses memory from a pool
+-- | and may contain sensitive data. See the Node docs.
```
Comment (thomashoneyman) [inline src/Node/Buffer.purs:104]: Worth providing a link to the Node docs for this, or is there not an easily-linkable section (same for the rest of the file where you've commented about Node docs)?

File: src/Node/Buffer/ST.purs [hunk 2]
```diff
@@ -74,10 +83,23 @@ instance mutableBufferST :: MutableBuffer (STBuffer h) (ST h) where
   copy = copy
   fill = fill
 
+-- | Creates a new `STBuffer`. Alias of `alloc`.
 create :: forall h. Int -> ST h (STBuffer h)
-create = unsafeCoerce Buffer.create
```
Comment (thomashoneyman) [inline src/Node/Buffer/ST.purs:78]: Are these exact replacements, in that someone who was previously using `create` will not be affected by a change to `alloc` underneath?

Comment (JordanMartinez) [reply src/Node/Buffer/ST.purs:78]: The [original FFI](https://github.com/purescript-node/purescript-node-buffer/blob/v8.0.0/src/Node/Buffer/Immutable.js#L17-L19) is using `alloc` underneath. However, from the name alone, it's not clear which Node API is being used. 

I made `create` an alias of `alloc` to reduce breakage. I think I'd prefer to remove it entirely, but that can be discussed in a separate PR as there are other cases like that.

File: src/Node/Buffer.purs (hunk 1 above)
Comment (JordanMartinez) [reply src/Node/Buffer.purs]: Done.

File: src/Node/Buffer/ST.purs (hunk 2 above)
Comment (thomashoneyman) [reply src/Node/Buffer/ST.purs:78]: I see. Thanks for clarifying.

**Comment by thomashoneyman** [comment]
Since we're pointing to Node v18 here, what do you think of updating CI to point to Node 18 (or perhaps to just the Node LTS, which is Node 18 and will become Node 20 soon)?

https://github.com/purescript-node/purescript-node-buffer/blob/c6533da65c7eff9677a2e6421004dd242ea3dc8b/.github/workflows/ci.yml#L20-L22

**Comment by JordanMartinez** [comment]
Sure.

**Pull request merged**